dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```diff
@@ -578,6 +595,39 @@ public static Dictionary<string, string> BuildPropertyNameToStorageNameMap((Vect
         return storagePropertyNameMap;
     }
 
+    /// <summary>
+    /// Verify if the provided type is one of the supported Enumerable types.
+    /// </summary>
+    /// <param name="type">The type to check.</param>
+    /// <returns><see langword="true"/> if the type is a supported Enumerable, <see langword="false"/> otherwise.</returns>
+    public static bool IsSupportedEnumerableType(Type type)
+    {
+        if (type.IsArray || type == typeof(IEnumerable))
+        {
+            return true;
+        }
+
+        if (typeof(IList).IsAssignableFrom(type) && type.GetConstructor([]) != null)
+        {
+            return true;
+        }
+
+        if (type.IsGenericType)
+        {
+            var genericTypeDefinition = type.GetGenericTypeDefinition();
+            if (genericTypeDefinition == typeof(ICollection<>) ||
+                genericTypeDefinition == typeof(IEnumerable<>) ||
+                genericTypeDefinition == typeof(IList<>) ||
+                genericTypeDefinition == typeof(IReadOnlyCollection<>) ||
+                genericTypeDefinition == typeof(IReadOnlyList<>))
```

Comment:
All of those are `IEnumerable<>`, just need to check if it implements `IEnumerable`.

IReadOnlyList -> IReadOnlyCollection -> IEnumerable 
IList -> ICollection -> IEnumerable

```suggestion
            if (genericTypeDefinition is IEnumerable<>)
```

Reply:
Yes, those are all inheriting from IEnumerable<T>, but I don't support all that inherit from IEnumerable<T>.
E.g. ISet<T> is not supported, but it inherits from IEnumerable<T>.  If someone has IEnumerable<T> on their data model, I can support by creating a List<T> and passing it as an IEnumerable<T>, and same for ICollection<T>.  But I don't have support for every type in the same way.